Added create, read, update, delete operations for MockRecord

wasif-raza opened the pull request:
- Automatically sets expiresAt to 7 days from creation; removed expiresAt from client DTOs
- Implemented bulk record creation endpoint
- Updated service and repository to handle schema associations correctly
- Removed manual expiresAt handling from update to ensure consistent expiration

wasif-raza committed:
Added create, read, update, delete operations for MockRecord
- Automatically sets expiresAt to 7 days from creation; removed expiresAt from client DTOs
- Implemented bulk record creation endpoint
- Updated service and repository to handle schema associations correctly
- Removed manual expiresAt handling from update to ensure consistent expiration

arshadjama1 reviewed (changes requested):
### Define the Correct MockRecordService Interface
And then update your implementations of service and controller accordingly.

public interface MockRecordService {

    MockRecordResponse createRecord(Long userId, CreateMockRecordRequest request);

    List<MockRecordResponse> createRecordsBulk(Long userId, List<CreateMockRecordRequest> requests);

    MockRecordResponse getRecordById(Long userId, Long recordId);

    List<MockRecordResponse> getRecordsBySchemaId(Long userId, Long schemaId);

    MockRecordResponse updateRecord(Long userId, Long recordId, UpdateMockRecordRequest request);

    void deleteRecord(Long userId, Long recordId);

    void deleteExpiredRecords(); // system-level, no userId needed

    long countRecords();
}

wasif-raza committed:
- Added @Slf4j annotation
- Added info/debug/warn logs for create, update, delete, bulk operations
- Added validation logs for record schema checks
- Improved visibility into service workflow and error points

arshadjama1 reviewed (changes requested):
Please update your MockRecord service interface and its implementations and uses in the controller. 

> As asked in earlier request.

wasif-raza committed:
- Added userId-based permission check to ensure only the organization owner can create schemas
- Implemented owner validation across project → organization → user relationships
- Updated service interfaces to include userId parameter for secure access flows
- Improved access control consistency across controller, service, and interface layers
- Ensures schemas can only be created/managed by the correct authenticated user

arshadjama1 committed:
Refactor access control checks and validation checks

arshadjama1 committed:
Remove deleteExpiredRecords method from service interface

arshadjama1 committed:
Remove getOrganization method from MockSchema

Removed the getOrganization method from MockSchema.

arshadjama1 committed:
Modify endpoints and removed the countRecords method.
